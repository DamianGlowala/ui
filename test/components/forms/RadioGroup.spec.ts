import { describe, it, expect } from 'vitest'
import { URadioGroup } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('RadioGroup', () => {
  it.each([
    ['basic case', {}],
    ['with name and legend', { props: { name: 'radioGroupName', legend: 'Radio Group Legend' } }],
    ['with options and legend', { props: { options: [{ label: 'Option 1', value: 'option1' }, { label: 'Option 2', value: 'option2' }], legend: 'Radio Group Legend' } }],
    ['with optionAttribute and legend', { props: { optionAttribute: 'name', legend: 'Radio Group Legend' } }],
    ['with valueAttribute and legend', { props: { valueAttribute: 'id', legend: 'Radio Group Legend' } }],
    ['with disabled and legend', { props: { disabled: true, legend: 'Radio Group Legend' } }],
    ['with custom color and legend', { props: { color: 'blue', legend: 'Radio Group Legend' } }],
    ['with custom class and legend', { props: { class: 'custom-class', legend: 'Radio Group Legend' } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof RadioGroup.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, URadioGroup)
    expect(html).toMatchSnapshot()
  })
})
