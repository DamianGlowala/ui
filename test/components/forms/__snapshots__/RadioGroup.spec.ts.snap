// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadioGroup > renders basic case correctly 1`] = `
"<div class="">
  <fieldset>
    <!--v-if-->
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with custom class and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with custom color and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with disabled and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with modelValue and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with name and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with optionAttribute and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with options and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="Ooz6KnLRIo_0" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-primary-500 dark:text-primary-400" value="option1"></div>
      <div class="ms-3 flex flex-col"><label for="Ooz6KnLRIo_0" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 1
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="Ooz6KnLRIo_1" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-primary-500 dark:text-primary-400" value="option2"></div>
      <div class="ms-3 flex flex-col"><label for="Ooz6KnLRIo_1" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 2
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with valueAttribute and legend correctly 1`] = `
"<div class="">
  <fieldset>
    <legend class="">Radio Group Legend</legend>
  </fieldset>
</div>"
`;
